projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6670,6 +6676,14 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
             this.calculateGridWidth();
             this.cdr.detectChanges();
         }
+
+        // in case horizontal scrollbar has appeared recalc to size correctly.
+        if (hasHScroll !== this.hasHorizontalScroll()) {
+            this.isHorizontalScrollHidden = !this.hasHorizontalScroll();
+            this.cdr.detectChanges();
+            this.calculateGridHeight();
```

Comment:
Why do we need to calculate the grid height again when the horizontal scroll appears?

Reply:
Because the horizontal scrollbar takes up some of the vertical space when it show/hides.

Reply:
Is this fixing a specific issue because when resizing the resize observer triggers the calculation of the grid height couple of times and it seems ok to me?

Reply:
Yes. If you go from a grid that has horizontal scrollbar to a grid that should have no horizontal scrollbar (for example going from a minimized window and them maximizing it so that all columns fit) then the horizontal scrollbar container will not hide without the additional handling. There were some tests that were failing due to this.

Wrong behavior, can be observed for example here:
![horizontalNoHide](https://github.com/IgniteUI/igniteui-angular/assets/10397980/8bf56d8f-0240-4837-b481-b14428ce5edb)

Notice that the horizontal scrollbar is no longer needed when maximized, but still shows.

Correct behavior would be for it to hide fully:
 
![horizontalHide](https://github.com/IgniteUI/igniteui-angular/assets/10397980/69599cc5-f97b-4dbe-bec3-a32ae435f0e2)